File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.js
/**
 * @param {number[]} salary
 * @return {number}
 */
var average = function(salary) {
    let total = 0
    let max = salary[0]
    let min = salary[0]
    for(let i = 0; i < salary.length; i++){
        if (salary[i] >= max){
            max = salary[i]
        }else if(salary[i] <= min){
            min = salary[i]
        }
        total += salary[i] 
    }
    return (total-max-min) / (salary.length -2)
};

File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.java
class Solution {
    public double average(int[] salary) {
       Arrays.sort(salary);
        double total = 0;
        for (int i = 1; i < salary.length -1;i++){
            total += salary[i];
        }
        return (double) (total / (salary.length - 2));  
    }
}